test/IntegrationTests/SmokeTests.cs

```diff
@@ -271,8 +273,84 @@ public void LogsNoneInstrumentations()
 
         collector.AssertEmpty(5.Seconds());
     }
+
 #endif
 
+    [Fact]
+    [Trait("Category", "EndToEnd")]
+    public void TracesNoneInstrumentations()
+    {
+        using var collector = new MockSpansCollector(Output);
+        SetEnvironmentVariable("OTEL_DOTNET_AUTO_TRACES_ENABLED_INSTRUMENTATIONS", "none");
+        RunTestApplication();
+        collector.AssertEmpty(1.Seconds());
+    }
+
+    [Fact]
+    [Trait("Category", "EndToEnd")]
+    public void MetricsNoneInstrumentations()
+    {
+        using var collector = new MockMetricsCollector(Output);
+        SetEnvironmentVariable("OTEL_DOTNET_AUTO_METRICS_ENABLED_INSTRUMENTATIONS", "none");
+        RunTestApplication();
+        collector.AssertEpmty(1.Seconds());
+    }
+
+    [Fact]
+    [Trait("Category", "EndToEnd")]
+    public void LogsDisabledInstrumentation()
+    {
+        using var collector = new MockLogsCollector(Output);
+        SetEnvironmentVariable("OTEL_DOTNET_AUTO_LOGS_DISABLED_INSTRUMENTATIONS", "ILogger");
+        EnableBytecodeInstrumentation();
+        RunTestApplication();
+        collector.AssertEmpty(1.Seconds());
+    }
+
+    [Fact]
+    [Trait("Category", "EndToEnd")]
+    public void MetricsDisabledInstrumentation()
+    {
+        using var collector = new MockMetricsCollector(Output);
+        SetExporter(collector);
+
+        StringBuilder sbToDisableMetrics = new StringBuilder();
+        sbToDisableMetrics.Append(MetricInstrumentation.NetRuntime.ToString())
+        .Append(",").Append(MetricInstrumentation.HttpClient.ToString());
+
+        SetEnvironmentVariable("OTEL_DOTNET_AUTO_METRICS_DISABLED_INSTRUMENTATIONS", sbToDisableMetrics.ToString());
+        EnableBytecodeInstrumentation();
+        RunTestApplication();
+        Assert.False(
+            collector.CheckInstrumentationScopeIsAbsent(
+            new List<string>() { "OpenTelemetry.Instrumentation.Process" },
+            1.Seconds()),
+            "Didn't find metrics instrumentation while it should be present.");
+        Assert.True(
+            collector.CheckInstrumentationScopeIsAbsent(
+            new List<string>() { "OpenTelemetry.Instrumentation.NetRuntime", "OpenTelemetry.Instrumentation.HttpClient" },
+            1.Seconds()),
+            "Found Metrics instrumentation while it should be disabled.");
+    }
+
+    [Fact]
+    [Trait("Category", "EndToEnd")]
+    public void TracesDisabledInstrumentation()
+    {
+        using var collector = new MockSpansCollector(Output);
+        SetExporter(collector);
+
+        StringBuilder sbToDisableMetrics = new StringBuilder();
+        sbToDisableMetrics.Append(TracerInstrumentation.HttpClient.ToString())
+        .Append(",").Append(TracerInstrumentation.AspNet.ToString());
+
+        SetEnvironmentVariable("OTEL_DOTNET_AUTO_TRACES_DISABLED_INSTRUMENTATIONS", sbToDisableMetrics.ToString());
```

Comment:
Clever (really!). However, I would prefer to have it simple as I think it would be more readable to have

```suggestion
        SetEnvironmentVariable("OTEL_DOTNET_AUTO_TRACES_DISABLED_INSTRUMENTATIONS", "AspNet,HttpClient");
```

This is an integration test which is more on the "user level". The env vars changes should not be threated lightly as this is our user facing interface.

Also if the env var value would change the enum would change anyway 😉